src/app/games/hive/hive.component.ts

```diff
@@ -25,9 +25,60 @@ import { HiveRules } from './HiveRules';
 import { HiveState } from './HiveState';
 import { HiveTutorial } from './HiveTutorial';
 
-interface StackAndCoord {
-    stack: HivePieceStack;
-    coord: Coord;
+interface GroundInfo {
+    spaceClasses: string[];
+    strokeClasses: string[];
+}
+
+class Ground extends Table2DWithPossibleNegativeIndices<GroundInfo> {
+    private highlighted: Coord[] = [];
+
+    public initialize(coord: Coord): void {
+        this.set(coord, { spaceClasses: [], strokeClasses: [] });
+    }
+    public highlightFill(coord: Coord, fill: string): void {
+        this.highlighted.push(coord);
+        this.get(coord).map((g: GroundInfo) => g.spaceClasses.push(fill));
```

Comment:
this.get(coord).spaceClasses.push(fill) ?

Reply:
same pour highlightStroke

Reply:
same pour clearHighlights ?

Reply:
mais c'point la même chose ! 
this.get(coord): MGPOptional<GroundInfo>
Si tu veux dire
`this.get(coord).get().spaceClasses.push(fill)`
C'point le même code, celui-ci throw si y'a rien où on est !
On peut faire un ifPresent:
```
const bli: MGPOptional<GroundInfo> = this.get(coord);
if (bli.isPresent()) {
    bli.get().spaceClasses.push(fill);
}
````
mais meh